astropy/io/ascii/daophot.py

```diff
@@ -12,15 +12,84 @@
 
 import re
 import numpy as np
+import itertools as itt
+from collections import defaultdict
+
 from . import core
-from . import basic
 from . import fixedwidth
 from ...utils import OrderedDict
+from ...extern.six.moves import zip, map
+from .misc import first_true_index, first_false_index, groupmore
 
 
 class DaophotHeader(core.BaseHeader):
-    comment = r'\s*#K'
-    aperture_values = ''
+    """Read the header from a file produced by the IRAF DAOphot routine."""
+
+    def __init__(self):
+        core.BaseHeader.__init__(self)
+        self.comment = r'\s*#K'
```

Comment:
I strongly prefer to define these things as class attributes, don't just assign them in `__init__`. That makes it harder to subclass this class (we had a earlier issue where I moved definitions like this from __init__ up one level (I just don't remember the number of that PR now).
The same applies to re_format, re_header_keyword etc.